### ISSUE TITLE
Empty call stack when the function is referenced in an other way than by calling

### ISSUE THREAD

**Issue by lenticularis39** [viktormalik/diffkemp#27]
**Example:**

A difference in the function `fcoe_ctlr_vn_age` is shown when comparing the KABI function `fcoe_ctlr_init`. The origin of the reference to the function is the `INIT_WORK` macro containing the function `fcoe_ctlr_timer_work`, which subsequently calls the differing function (`fcoe_ctlr_vn_age`):
```c
/* drivers/scsi/fcoe/fcoe_ctlr.c */
...
void fcoe_ctlr_init(struct fcoe_ctlr *fip, enum fip_state mode)
{
        ...
	INIT_WORK(&fip->timer_work, fcoe_ctlr_timer_work);
...
```
This results in (among other things) this `store` instruction in the LLVM IR:
```llvm
; drivers/scsi/fcoe/fcoe_ctlr.ll
...
define void @fcoe_ctlr_init(%struct.fcoe_ctlr* %fip, i32 %mode) #0 !dbg !2 {
        ...
        store void (%struct.work_struct*)* @fcoe_ctlr_timer_work, void (%struct.work_struct*)** %func, align 8, !dbg !11060, !tbaa !11062
        ...
}
```
The `@fcoe_ctlr_timer_work` gets compared using the `cmpGlobalValues` function, which triggers the code in `DifferentialGlobalNumberState`, comparing the function, eventually getting to the function with the reported difference and adding it to the list of functions to semantically compare.

The function is compared, but it wasn't found through a call instruction, therefore the call stack is empty, which is confusing.

**Proposed solutions:**

This can be fixed in multiple ways, depending on what the behaviour should be. One is to replace the call stack with a use backtrace showing all uses of the function, not only calls. Another solution would be not to compare these functions when comparing the control flow only. Or it is possible to somehow print that the function is used in another way than a call.

**Comment by lenticularis39** [comment]
This is apparently the only cause of empty call stacks (besides the function being the KABI function itself).

**Comment by viktormalik** [comment]
Maybe we could be able to find out whether the function is actually called (i.e. in this case, `%func` is called) and if not, we could ignore it when comparing the control flow only. Otherwise, we would include it into the callstack.

**Comment by viktormalik** [comment]
Fixed by e4e2e301eff15514efbbf2e6a804e1e436bd5225 and 5f8fb100c34748f64b95b21f8b672583c0a49eb9.
